### PR TITLE
REPLACE language directories with index directory

### PR DESCRIPTION
**Pull request by alexmoroianu**

### CODE REVIEW

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import app.coronawarn.server.services.distribution.assembly.structure.file.FileOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig.DigitalGreenCertificate;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class DigitalCertificatesDirectory extends IndexDirectoryOnDisk<String> {
+
+  public static final String EXPORT_BIN = "export.bin";
```
Comment (roesslerj) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java:18]: I dislike this name, as it has to change if the contents changes. How about FILENAME or EXPORT_BINARY_FILENAME or something similar semantical?

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    this.cryptoProvider = cryptoProvider;
+  }
+
+  @Override
+  public void prepare(ImmutableStack<Object> indices) {
+    if (valueSets != null) {
+      this.addWritableToAll(ignoredValue -> {
+        ArchiveOnDisk archiveToPublish = new ArchiveOnDisk(dgcConfig.getValuesetsFileName());
+        archiveToPublish.addWritable(new FileOnDisk(EXPORT_BIN, valueSets.toByteArray()));
+        return Optional.of(new DistributionArchiveSigningDecorator(
```
Comment (roesslerj) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java:50]: I miss the logging statement we had before, at least in debug level...

**Review by roesslerj — CHANGES_REQUESTED** [review]

File: services/distribution/src/main/resources/application.yaml [hunk 3]
```diff
@@ -226,25 +226,24 @@ services:
...
+        max-retry-attempts: 2
+        base-url: ${BUSSINESS_RULES_BASE_PATH:https://distribution-dfe4f5c711db.dcc-rules.de/}
         country-list-path: ${BUSSINESS_RULES_COUNTRY_LIST_PATH:/countrylist}
         value-sets-path: ${BUSSINESS_RULES_VALUE_SETS_PATH:/valuesets}
         rules-path: ${BUSSINESS_RULES_RULES_PATH:/rules}
         ssl:
-          trust-store: ${DCC_TRUST_STORE:../../docker-compose-test-secrets/dcc_truststore}
-          trust-store-password: ${DCC_TRUST_STORE_PASSWORD:123456}
+          trust-store: ../../docker-compose-test-secrets/dcc_truststore
+          trust-store-password: 123456
```
Comment (ioangut) [inline services/distribution/src/main/resources/application.yaml:247]: Don't we use anymore thee DCC_TRUST_STORE and DCC_TRUST_STORE_PASSWORD as env variable? (this is application.yaml from mail not test )

Comment (alexmoroianu) [reply services/distribution/src/main/resources/application.yaml]: Good catch :D Modified it.

Comment (alexmoroianu) [reply services/distribution/src/main/resources/application.yaml]: Good catch. I updated it :D

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java (hunk 2 above)
Comment (alexmoroianu) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java:56]: I added the logger :D

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java (hunk 1 above)
Comment (alexmoroianu) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java]: Right. Updated

**Review by roesslerj — APPROVED** [review]

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  private final DistributionServiceConfig distributionServiceConfig;
+
+  private final DigitalGreenCertificate dgcConfig;
+
+  private final CryptoProvider cryptoProvider;
+
+  private ValueSets valueSets;
+
+  /**
+   * Constructs a {@link IndexDirectoryOnDisk} instance that represents a directory, containing an index in the form of
```
Comment (FelixRottler) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java:33]:
```suggestion
   * Constructs a {@link IndexDirectoryOnDisk} instance that represents a directory for a digital green certificates and writes valuesets to all supported countries by the dgc.
```

Maybe we can have a more meaningful comment. Sry for nitpicking here

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/ValueSetsNotFetchedfStructureProviderTest.java [hunk 5]
```diff
@@ -97,10 +97,12 @@ void should_not_contain_valuesets_if_any_is_not_fetched() throws FetchValueSetsE
     digitalGreenCertificates.prepare(new ImmutableStack<>());
 
     assertEquals("ehn-dgc", digitalGreenCertificates.getName());
-    List<String> supportedLanguages = digitalGreenCertificates.getWritables().stream().map(Writable::getName).collect(
-        Collectors.toList());
     List<String> expectedLanguages = Arrays.asList("de", "en", "bg", "pl", "ro", "tr");
-    assertTrue(supportedLanguages.stream().noneMatch(expectedLanguages::contains));
+    boolean areLanguageFoldersEmpty = digitalGreenCertificates.getWritables().stream()
```
Comment (FelixRottler) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/ValueSetsNotFetchedfStructureProviderTest.java:101]: Type in the class name: ValueSetsNotFetched**f**StructureProviderTest

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/ValueSetsNotFetchedfStructureProviderTest.java [hunk 6]
```diff
@@ -97,10 +97,12 @@ void should_not_contain_valuesets_if_any_is_not_fetched() throws FetchValueSetsE
...
+        .filter(writableOnDiskWritable -> expectedLanguages.contains(writableOnDiskWritable.getName()))
+        .filter(writableOnDiskWritable -> writableOnDiskWritable instanceof DirectoryOnDisk)
+        .noneMatch(directory -> ((DirectoryOnDisk) directory).getWritables().size() > 0);
```
Comment (FelixRottler) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/ValueSetsNotFetchedfStructureProviderTest.java:104]:
```suggestion
        .allMatch(directory -> ((DirectoryOnDisk) directory).getWritables().isEmpty());
```

**Review by FelixRottler — CHANGES_REQUESTED** [review]
See comments :)

**Review by FelixRottler — APPROVED** [review]

**Pull request merged**